libvips/arithmetic/find_trim.c

```diff
@@ -92,13 +92,13 @@ vips_find_trim_build(VipsObject *object)
 	/* Is "background" unset? Default to the correct value
 	 * for this interpretation.
 	 */
-	if (!vips_object_argument_isset(object, "background"))
-		if (find_trim->in->Type == VIPS_INTERPRETATION_GREY16 ||
-			find_trim->in->Type == VIPS_INTERPRETATION_RGB16) {
+	if (!vips_object_argument_isset(object, "background")) {
+		double max_val = vips_interpretation_max_alpha(find_trim->in->Type);
+		if (max_val != 255.0) {
```

Comment:
How about not setting `background` in `vips_find_trim_init()` and leaving it NULL? You could remove this `if()` and get rid of the 255 magic number.